backend/users/src/main/java/org/eclipse/sw360/users/UserHandler.java

```diff
@@ -44,13 +50,40 @@ public class UserHandler implements UserService.Iface {
 
     private static final Logger log = LogManager.getLogger(UserHandler.class);
     private static final String EXTENSION = ".log";
+    private final PasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
 
     private UserDatabaseHandler db;
     private ReadFileDepartmentConfig readFileDepartmentConfig;
 
     public UserHandler() throws IOException {
-        db = new UserDatabaseHandler(DatabaseSettings.getConfiguredClient(), DatabaseSettings.COUCH_DB_USERS);
+        db = new UserDatabaseHandler(DatabaseSettings.getConfiguredClient(),
+                DatabaseSettings.COUCH_DB_USERS);
         readFileDepartmentConfig = new ReadFileDepartmentConfig();
+
+        // Create admin user if not in database yet
```

Comment:
Just a suggestion to add it inside `Sw360UserService.java` or `UserController.java` 
```
   @PostConstruct
    private void initializeDefaultAdminUser() {
        createAdminUserIfNotExists();
    }

    private void createAdminUserIfNotExists() {
        // Create admin user if not in database yet
        List<User> users = getAllUsers();
        if (users.isEmpty()) {
            Optional<String> COUCHDB_ADMIN_PASSWORD = Optional.ofNullable(
                    System.getenv("COUCHDB_ADMIN_PASSWORD") != null
                            ? System.getenv("COUCHDB_ADMIN_PASSWORD")
                            : "sw360fossie"
            );
            User admin = new User();
            admin.setEmail("setup@sw360.org");
            admin.setFullname("SW360 Admin");
            admin.setGivenname("SW360");
            admin.setLastname("Admin");
            admin.setDepartment("SW360");
            admin.setPassword(COUCHDB_ADMIN_PASSWORD.get());
            admin.setUserGroup(UserGroup.ADMIN);
            String encodedPassword = passwordEncoder.encode(
                    admin.getPassword()
            );
            admin.setPassword(encodedPassword);
            log.info("No users found. Creating default administrator user.");
            try {
                addUser(admin);
            } catch (TException e) {
                log.atError().withThrowable(e).log("Error creating admin user");
            }
        }
```

Reply:
No dice. Thrift is set in a way that waits all services be available before enable communication, and we're trying to use it befor even been enabled.
Tests show that the startup of resource server becomes stuck.
So, the current imlpementation is the best until we in the future remove Thrift.